test/Generator/ParameterGeneratorTest.php

```diff
@@ -370,11 +407,40 @@ public function internalReflectionHintsProvider()
-            [ClassTypeHintedClass::class, 'selfParameter', 'foo', ClassTypeHintedClass::class],
-            [ClassTypeHintedClass::class, 'classParameter', 'foo', ClassTypeHintedClass::class],
-            [ClassTypeHintedClass::class, 'otherClassParameter', 'foo', InternalHintsClass::class],
-            [ClassTypeHintedClass::class, 'closureParameter', 'foo', \Closure::class],
-            [ClassTypeHintedClass::class, 'importedClosureParameter', 'foo', \Closure::class],
+            [NullableHintsClass::class, 'arrayParameter', 'foo', '?array'],
+            [NullableHintsClass::class, 'callableParameter', 'foo', '?callable'],
+            [NullableHintsClass::class, 'intParameter', 'foo', '?int'],
+            [NullableHintsClass::class, 'floatParameter', 'foo', '?float'],
+            [NullableHintsClass::class, 'stringParameter', 'foo', '?string'],
+            [NullableHintsClass::class, 'boolParameter', 'foo', '?bool'],
+            [NullableHintsClass::class, 'selfParameter', 'foo', '?\\' . NullableHintsClass::class],
+            [NullableHintsClass::class, 'parentParameter', 'foo', '?\\' . EmptyClass::class],
+            [NullableHintsClass::class, 'nullableHintsClassParameter', 'foo', '?\\' . NullableHintsClass::class],
+            [NullNullableDefaultHintsClass::class, 'arrayParameter', 'foo', '?array'],
+            [NullNullableDefaultHintsClass::class, 'callableParameter', 'foo', '?callable'],
+            [NullNullableDefaultHintsClass::class, 'intParameter', 'foo', '?int'],
+            [NullNullableDefaultHintsClass::class, 'floatParameter', 'foo', '?float'],
+            [NullNullableDefaultHintsClass::class, 'stringParameter', 'foo', '?string'],
+            [NullNullableDefaultHintsClass::class, 'boolParameter', 'foo', '?bool'],
+            [
+                NullNullableDefaultHintsClass::class,
+                'selfParameter',
+                'foo',
+                '?\\' . NullNullableDefaultHintsClass::class,
+            ],
+            [NullNullableDefaultHintsClass::class, 'parentParameter', 'foo', '?\\' . EmptyClass::class],
+            [
+                NullNullableDefaultHintsClass::class,
+                'nullableDefaultHintsClassParameter',
+                'foo',
+                '?\\' . NullNullableDefaultHintsClass::class,
+            ],
+            [ClassTypeHintedClass::class, 'selfParameter', 'foo', '\\' . ClassTypeHintedClass::class],
+            [ClassTypeHintedClass::class, 'parentParameter', 'foo', '\\' . EmptyClass::class],
+            [ClassTypeHintedClass::class, 'classParameter', 'foo', '\\' . ClassTypeHintedClass::class],
+            [ClassTypeHintedClass::class, 'otherClassParameter', 'foo', '\\' . InternalHintsClass::class],
+            [ClassTypeHintedClass::class, 'closureParameter', 'foo', '\\' . \Closure::class],
+            [ClassTypeHintedClass::class, 'importedClosureParameter', 'foo', '\\' . \Closure::class],
```

Comment:
Out of curiousity... why are the `Null*` cases prior to the others, which are all in alpha order by class name otherwise? (I ask, because I thought at first you'd removed the `ClassTypeHintedClass` cases until I discovered they'd been shifted down...)